cuppa/src/main/java/org/forgerock/cuppa/functions/Condition.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 ForgeRock AS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.forgerock.cuppa.functions;
+
+/**
+ * A condition that needs to be met.
+ */
+@FunctionalInterface
+public interface Condition {
```

Comment:
Could we use `java.util.function.Predicate` instead of creating a very similar interface?

Reply:
Predicate requires an input parameter, which we don't really have in case of a test definition.

Reply:
Ah yes, of course. How about `BooleanSupplier` then?